File: typeorm/src/entity/Order.ts
import {Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn, OneToMany} from "typeorm";
import { LineItem } from "./LineItem";
import { Payment } from "./Payment";

@Entity("orders")
export class Order {

    @PrimaryGeneratedColumn()
    id: number;

    @OneToMany(type => LineItem, lineItem => lineItem.order)
    lineItems: LineItem[];

    @OneToMany(type => Payment, payment => payment.order)
    payments: Payment[];
}

File: spring/src/main/java/learning/appointmentapp/entities/Order.java
package learning.appointmentapp.entities;

import java.util.Set;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;

@Entity
@Table(name = "orders")
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToMany(mappedBy = "order")
    private Set<LineItem> lineItems;

    @OneToMany(mappedBy = "order")
    private Set<Payment> payments;

    public Long getId() {
        return this.id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Set<LineItem> getLineItems() {
        return this.lineItems;
    }

    public void setLineItems(Set<LineItem> lineItems) {
        this.lineItems = lineItems;
    }

    public Set<Payment> getPayments() {
        return this.payments;
    }

    public void setPayments(Set<Payment> payments) {
        this.payments = payments;
    }

}